### PR TITLE
Feature/fix ioexception

### PR DESCRIPTION
**Pull request by idodeclare**
Hello,

Please consider for integration this patch to fix an IOException that is occurring for code files larger than 8192 bytes in size. I created a sample, a.csproj; added a (breaking) test in the first commit; and then fixed it in the second.

Thank you.

### CODE REVIEW

File: src/org/opensolaris/opengrok/analysis/AnalyzerGuru.java [hunk 1]
```diff
@@ -125,7 +125,14 @@
      * Define very close to OPENING_MAX_CHARS so the input stream stays easily
      * in the <code>mark</code> limit
      */
-    private static final int OPENING_BUF_SIZE = 102;
+    private static final int OPENING_BUF_SIZE = 128;
```
Comment (vladak) [inline src/org/opensolaris/opengrok/analysis/AnalyzerGuru.java]: why is this being bumped to 128 ?

File: test/org/opensolaris/opengrok/analysis/AnalyzerGuruTest.java [hunk 2]
```diff
@@ -244,6 +242,15 @@ public void getAnalyzerFactoryClass() {
     }
 
     @Test
+    public void shouldNotThrowGettingCsprojOpening() throws IOException {
+        InputStream res = getClass().getClassLoader().getResourceAsStream(
+            "org/opensolaris/opengrok/analysis/a.csproj");
```
Comment (vladak) [inline test/org/opensolaris/opengrok/analysis/AnalyzerGuruTest.java]: I wonder if we should stick the test files under subdirectory or elsewhere. There does not seem to be a convention for tests yet.

**Comment by vladak** [comment]
This is kind of sad bug (given the conditions it wouldn't be fair to call the fix likewise). `addFile()` constructs the stream like this:

```
        try (InputStream in = new BufferedInputStream(new FileInputStream(file))) {
            fa = AnalyzerGuru.getAnalyzer(in, path);
        }
```

and this gets eventually passed to `readOpening()`. So the chain of streams is FileInputStream -> BufferedInputStream -> InputStream and the chain of readers is InputStreamReader -> BufferedReader.

Now, BufferedReader has its own buffer as well as BufferedInputStream. `readOpening()` reads the data one character at a time, however the underlying BufferedReader reads by its own buffer size and in turn BufferedInputStream uses its own buffer size that advances the position in the output stream. So even though max 100 characters (`OPENING_MAX_CHARS`) can be read by the cycle in reality it could be 8K characters.

The last constant seems to be unreachable from  `readOpening()` (https://stackoverflow.com/a/42572182)

File: src/org/opensolaris/opengrok/analysis/AnalyzerGuru.java [hunk 3]
```diff
@@ -125,7 +125,14 @@
      * Define very close to OPENING_MAX_CHARS so the input stream stays easily
      * in the <code>mark</code> limit
```
Comment (vladak) [inline src/org/opensolaris/opengrok/analysis/AnalyzerGuru.java]: given this bug this comment is not very relevant. What about unifying OPENING_MAX_CHARS with OPENING_BUF_SIZE ?

Comment (idodeclare) [reply src/org/opensolaris/opengrok/analysis/AnalyzerGuru.java]: You're right. It's not useful as a distinct value. I pushed a removal.

File: test/org/opensolaris/opengrok/analysis/AnalyzerGuruTest.java (hunk 2 above)
Comment (idodeclare) [reply test/org/opensolaris/opengrok/analysis/AnalyzerGuruTest.java]: I thought about that — but a lower-cased file name does at least show it at the bottom of the test/org folder in NetBeans.

Comment (vladak) [reply test/org/opensolaris/opengrok/analysis/AnalyzerGuruTest.java]: Yeah, no big deal. We should probably rethink and change this for all tests.

**Pull request merged**